frb_codegen/src/commands.rs

```diff
@@ -301,6 +364,58 @@ fn guess_toolchain(dart_root: &str) -> anyhow::Result<DartToolchain> {
     Ok(DartToolchain::Dart)
 }
 
+fn ensure_dependencies(dart_root: &str, key: &str) -> anyhow::Result<Option<VersionReq>> {
+    debug!(
+        "Checking presence of {} in dependencies in {}",
+        key, dart_root
+    );
+    let manifest_file = read_file(dart_root, "pubspec.yaml")?;
+    info!("{:#?}", manifest_file);
+    let manifest_file: Pubspec = serde_yaml::from_str(&manifest_file).map_err(|e| {
+        info!("{}", e);
+        anyhow::Error::msg(format!("unable to parse pubspec.yaml in {}", dart_root))
+    })?;
+    let deps = manifest_file.dependencies.unwrap_or_default();
+    let version = deps.get(key);
+    let version = version.map(|v| match v {
+        DependencyVersion::Inline(ref version) => VersionReq::parse(version).unwrap(),
+        DependencyVersion::Multiline { ref version } => VersionReq::parse(
+            version
+                .as_ref()
+                .expect(&format!("please specify a version for {}", key)),
+        )
+        .unwrap(),
+    });
+    info!("found version {:#?} for {}", version, key);
+    Ok(version)
+}
+
+fn ensure_dev_dependencies(dart_root: &str, key: &str) -> anyhow::Result<Option<VersionReq>> {
```

Comment:
Oops ensure_dependencies and ensure_dev_dependencies seems to be so duplicated. I guess they only differ in a string - "dependencies" vs "dev_dependencies" key in yaml

Reply:
It should be ok 👌